src/app/main-page/presentation/toolbar-action/toolbar-action.component.ts

```diff
@@ -128,14 +130,29 @@ export class ToolbarActionComponent implements OnChanges {
         this.graphType.emit(this.graphSettings.type);
     }
 
-    setMapType(mapType: MapType) {
-        this.mapType.emit(mapType);
+    setMapSettings(userInput: MapType | TileServer): void {
```

Comment:
why don't you choose 
```
setMapType(mapType: MapType): void {
  this.mapSettings.emit({ mapType: mapType });
}

setTileServer(tileServer: TileServer): void {
  this.mapSettings.emit({ mapType: MapType.TILES_ONLY, tileServer: tileServer });
}
```

Reply:
- [ ] refactor to use two setters but only one emit

Reply:
fixed